src/posix.c

```diff
@@ -216,9 +217,48 @@ int p_write(git_file fd, const void *buf, size_t cnt)
 	return 0;
 }
 
-#ifdef NO_MMAP
+ssize_t p_pread(git_file fd, void *buf, size_t nbyte, git_off_t offset)
```

Comment:
If we have indirection between the `unix` and `win32` compatibility layers, they should go into a file in `unix/` and `win32/` directories (eg `win32/posix_w32.c`) so that we can avoid the ifdefs.